servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpServerBuilder.java

```diff
@@ -261,7 +262,11 @@ public HttpServerBuilder bufferAllocator(final BufferAllocator allocator) {
 
     @Override
     public Single<HttpServerContext> listenStreaming(final StreamingHttpService service) {
-        return listenForService(service, strategy);
+        HttpExecutionStrategy serviceStrategy = requiredOffloads(service, defaultStrategy());
+        HttpExecutionStrategy filterStrategy = computeRequiredStrategy(serviceFilters, serviceStrategy);
+        HttpExecutionStrategy useStrategy = defaultStrategy() == strategy ? offloadAll() :
+                strategy.hasOffloads() ? strategy.merge(filterStrategy) : offloadNone();
```

Comment:
Why does this have to be different from all other API variants? I mean, why don't we use `computeServiceStrategy` method here? Let's discuss and we should at least leave a comment in the code to clarify the difference.

My main concerning part is `defaultStrategy() == strategy ? offloadAll()`. If I'm a user who has 0 filters and passed `StreamingHttpService` impl that override `requireOffloads()` to `offloadNone()`, I won't expect `offloadAll()` by default.

Reply:
We can't use `computeServiceStrategy()` because it maps `defaultStrategy()` which we want to handle differently. The other three variants with adapters have their only handling for `defaultStrategy()` and won't ever trigger the `defaultStrategy() == strategy` in `computeServiceStrategy()`

> My main concerning part is `defaultStrategy() == strategy ? offloadAll()`. If I'm a user who has 0 filters and passed `StreamingHttpService` impl that override `requiredOffloads()` to `offloadNone()`, I won't expect `offloadAll()` by default.

Fair point. If the service does not override `requiredOffloads()` then the default `offloadAll()` it provides is sufficient to ensure safety otherwise builder `defaultStrategy()` allows whatever strategy the filter and service require.

With this change, and the additional change of returning `strategy` when `strategy.hasNoOffloads()` is true the result is that we can use `computeServiceStrategy()` everywhere.